search-provider/yelp-search-provider.c

```diff
@@ -0,0 +1,498 @@
+
+    g_variant_builder_init (&param_builder, G_VARIANT_TYPE ("av"));
+    g_variant_builder_add (&param_builder, "v", g_variant_new_string (result));
+
+    reply = g_dbus_connection_call_sync (session_bus,
+                                         "org.gnome.Yelp",
+                                         "/org/gnome/Yelp",
+                                         "org.freedesktop.Application",
+                                         "ActivateAction",
+                                         g_variant_new ("(sava{sv})",
+                                                        "show-page",
+                                                        &param_builder,
+                                                        NULL),
+                                         NULL,
+                                         G_DBUS_CALL_FLAGS_NONE,
+                                         -1,
+                                         NULL,
+                                         &error);
+
+    g_object_unref (session_bus);
+    if (reply)
+      g_variant_unref (reply);
+    if (error) {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      g_error_free (error);
+      return;
+    }
+    g_dbus_method_invocation_return_value (invocation, NULL);
+}
+
+static void
+search_provider_name_acquired_cb (GDBusConnection *connection,
+                                  const gchar     *name,
+                                  gpointer         user_data)
+{
+    g_debug ("Search provider name acquired: %s\n", name);
+}
+
+static void
+search_provider_name_lost_cb (GDBusConnection *connection,
+                              const gchar     *name,
+                              gpointer         user_data)
+{
+    g_debug ("Search provider name lost: %s\n", name);
+}
+
+static void
+search_provider_bus_acquired_cb (GDBusConnection *connection,
+                                 const gchar *name,
+                                 gpointer user_data)
+{
+    YelpSearchProviderApp *self = YELP_SEARCH_PROVIDER_APP (user_data);
+
+    self->object_manager = g_dbus_object_manager_server_new ("/org/gnome/Yelp/SearchProvider");
+    self->skeleton = yelp_shell_search_provider2_skeleton_new ();
+
+    g_signal_connect (self->skeleton, "handle-get-initial-result-set",
+                    G_CALLBACK (handle_get_initial_result_set), self);
+    g_signal_connect (self->skeleton, "handle-get-subsearch-result-set",
+                    G_CALLBACK (handle_get_subsearch_result_set), self);
+    g_signal_connect (self->skeleton, "handle-get-result-metas",
+                    G_CALLBACK (handle_get_result_metas), self);
+    g_signal_connect (self->skeleton, "handle-activate-result",
+                    G_CALLBACK (handle_activate_result), self);
+    g_signal_connect (self->skeleton, "handle-launch-search",
+                    G_CALLBACK (handle_launch_search), self);
+
+    g_dbus_interface_skeleton_export (G_DBUS_INTERFACE_SKELETON (self->skeleton),
```

Comment:
I would pass a GError instance here instead of `NULL` and check if this function returns FALSE. If so, I would add some g_warning about the fact that we failed to export the interface on the connection.